src/viadot/sources/eurostat.py

```diff
@@ -0,0 +1,290 @@
+
+        indicator_list = [eurostat_dictionary["value"][i] for i in index_list]
+        df.indicator = indicator_list
+        df[signal_lists[1].signal_name] = col_signal_temp
+        df[signal_lists[0].signal_name] = row_signal_temp
+
+        return df
+
+    @add_viadot_metadata_columns
+    def to_df(
+        self,
+        dataset_code: str,
+        params: dict = None,
+        base_url: str = "https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data/",
+        requested_columns: list = None,
+        if_empty: str = "fail",
+        tests: dict = None,
+    ) -> pd.DataFrame:
+        """Function responsible for getting response, creating DataFrame using method 'eurostat_dictionary_to_df'
+           with validation of provided parameters and their codes if needed.
+        Args:
+            dataset_code (str): The code of eurostat dataset that we would like to upload.
+            params (Dict[str], optional):
+                A dictionary with optional URL parameters. The key represents the parameter id, while the value is the code
+                for a specific parameter, for example: params = {'unit': 'EUR'} where "unit" is the parameter that you would like to set
+                and "EUR" is the code of the specific parameter. You can add more than one parameter, but only one code per parameter!
+                So you CAN NOT provide list of codes as in example 'params = {'unit': ['EUR', 'USD', 'PLN']}'
+                These parameters are REQUIRED in most cases to pull a specific dataset from the API.
+                Both parameter and code has to be provided as a string! Defaults to None.
+            base_url (str): The base URL used to access the Eurostat API. This parameter specifies the root URL for all requests made to the API.
+                It should not be modified unless the API changes its URL scheme.
+                Defaults to "https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data/"
+            requested_columns (List[str], optional): list of needed names of columns. Names should be given as str's into the list.
+                Defaults to None.
+            tests:
+                - `column_size`: dict{column: size}
+                - `column_unique_values`: list[columns]
+                - `column_list_to_match`: list[columns]
+                - `dataset_row_count`: dict: {'min': number, 'max', number}
+                - `column_match_regex`: dict: {column: 'regex'}
+                - `column_sum`: dict: {column: {'min': number, 'max': number}}
+        Raises:
+            APIError: If there is an error with the API request.
+            ValueError: If the resulting DataFrame is empty.
+            TypeError: If self.params is different type than a dictionary.
+            TypeError: If self.requested_columns is different type than a list containing strings.
+        Returns:
+            pd.DataFrame: Final DataFrame or raise prefect.logger.error, if issues occur.
```

Comment:
Remove mention of prefect, also I don't think this is accurate?

Reply:
Remmoved